docs/source/scripts/exportdb/generate_example_rst.py

```diff
@@ -0,0 +1,184 @@
+        )
+        graph_output = str(exported_program)
+        graph_output = re.sub(r"        # File(.|\n)*?\n", "", graph_output)
+        graph_output = graph_output.replace("\n", "\n    ")
+        output = f"    {graph_output}"
+    except torchdynamo.exc.Unsupported as e:
+        output = "    Unsupported: " + str(e).split("\n")[0]
+
+    doc_contents += output + "\n"
+
+    if len(splitted_source_code) == 2:
+        doc_contents += f"""\n
+You can rewrite the example above to something like the following:
+
+.. code-block:: python
+
+{splitted_source_code[1]}
+
+"""
+
+    # with open(os.path.join(EXPORTDB_SOURCE, f"{example_case.name}.rst"), "w") as f:
+    #     f.write(doc_contents)
+
+    return doc_contents
+
+
+def generate_index_rst(example_cases, tag_to_modules, support_level_to_modules):
+    """
+    Generates the index.rst file
+    """
+
+    # support_level_to_modules = {}
+    # for _, example_case in sorted(example_cases.items()):
+    #     support_modules = support_level_to_modules.setdefault(
+    #         example_case.support_level, []
+    #     )
+    #     support_modules.append(example_case.name)
+
+    support_contents = ""
+    for k, v in support_level_to_modules.items():
+        support_level = k.name.lower().replace("_", " ").title()
+        module_contents = "\n\n".join(v)
+        support_contents += f"""
+{support_level}
+{'-' * (len(support_level))}
+
+{module_contents}
+"""
+
+    tag_names = "\n    ".join(t for t in tag_to_modules.keys())
+
+    with open(os.path.join(PWD, "blurb.md"), "r") as file:
+        blurb = file.read()
+
+    # Generate contents of the .rst file
+    doc_contents = f"""ExportDB
+========
+
+{blurb}
+
+.. toctree::
+    :maxdepth: 1
+    :caption: Tags
+
+    {tag_names}
+
+{support_contents}
+"""
+
+    with open(os.path.join(EXPORTDB_SOURCE, "index.rst"), "w") as f:
+        f.write(doc_contents)
+
+
+def generate_tag_rst(tag_to_modules):
+    """
+    For each tag that shows up in each ExportCase.tag, generate an .rst file
+    containing all the examples that have that tag.
+    """
+
+    for tag, modules_rst in tag_to_modules.items():
+        doc_contents = f"{tag} tag\n{'=' * (len(tag) + 4)}\n"
+        doc_contents += "\n\n".join(modules_rst).replace("=", "-")
+
+        with open(os.path.join(EXPORTDB_SOURCE, f"{tag}.rst"), "w") as f:
+            f.write(doc_contents)
+
+
+def generate_rst():
+    if not os.path.exists(EXPORTDB_SOURCE):
+        os.makedirs(EXPORTDB_SOURCE)
+
+    example_cases = all_examples()
+    tag_to_modules = {}
+    support_level_to_modules = {}
+    for example_case in example_cases.values():
+
+        doc_contents = generate_example_rst(example_case)
+
+        for tag in example_case.tags:
+            tag_to_modules.setdefault(tag, []).append(doc_contents)
+
+        support_level_to_modules.setdefault(example_case.support_level, []).append(doc_contents)
+
+    generate_tag_rst(tag_to_modules)
```

Comment:
are tag rsts still being used?

Reply:
yeah! theyre linked at the top

Reply:
is there still dedicated pages for them? or are they inlined?

Reply:
yeah, there are dedicated pages